src/assisted_test_infra/test_infra/controllers/containerized_controller.py

```diff
@@ -30,6 +33,43 @@ def run(self, **kwargs):
 
         utils.run_container(container_name=self._name, image=self._image, flags=base_run_flags + self._extra_flags)
 
+    @property
+    def container_id(self):
+        # Cached container id when created.
+        if not self._is_running:
+            raise RuntimeError("Unable to get container id when container is not running ")
+
+        if self._container_id is None:
+            podman_filter_name = f"podman-remote ps -f name={self._name} --format json"
+            out = utils.run_command(podman_filter_name, shell=True)[0]
+            podman_id = json.loads(out)[0]['Id']
+            self._container_id = podman_id
+
+        return self._container_id
+
+    def write_to_container(self, file_path_container: str, content: list[str], append: bool = False) -> None:
+        # Write to file inside the container - update running container
+        container_id = self.container_id
+
+        redirect_append = ">>"
+        redirect_override = ">"
+        empty = ""
+        exec_container = f"podman-remote exec -it {container_id} /bin/sh -c"
+
+        if not append:
+            # Reset file content if not append
+            utils.run_command(f"{exec_container} 'echo {empty}{redirect_override}{file_path_container}'", shell=True)
```

Comment:
We will update the pxe script dynamically .
Example:
create pxe server > empty (because we dont know what should be the first script) and save time on creation.

Loop over nodes(Serial  -> we dont need multiple PXE cotainers :
get node image path
Create iscsi target for the image
Update pxe iqn -> here we add content to the script and override per node.

Reply:
We hold single PXE server per cluster and the boot sctipt is the cluster name